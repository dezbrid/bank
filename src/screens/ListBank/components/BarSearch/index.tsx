import React, {useState} from 'react';
import {View, Image, TextInput, TouchableOpacity} from 'react-native';
import {Nullable} from '@interfaces/generic';
import {filterByName, getOriginalList} from '@redux/bankSlice';
import {useAppDispatch} from '@redux/hooks';

import iconSerch from './assets/ic_search.png';
import clearIcon from './assets/ic_close.png';
import styles from './styles';

function BarSearch() {
  const dispatch = useAppDispatch();
  const [search, setSearch] = useState<string>('');
  let inputRef: Nullable<TextInput> = null;
  const handlePressClearSearch = () => {
    inputRef!.clear();
    setSearch('');
    dispatch(getOriginalList());
  };
  const handleChangeText = (text: string) => {
    dispatch(filterByName(text.toLowerCase()));
    setSearch(text);
  };
  return (
    <View style={[styles.container, styles.containterShadow]}>
      <Image
        source={iconSerch}
        resizeMode="contain"
        style={styles.iconSearch}
      />
      <TextInput
        ref={ref => (inputRef = ref)}
        autoComplete="off"
        autoCorrect={false}
        style={styles.inputStyle}
        placeholder={'Buscar por nombre'}
        onChangeText={handleChangeText}
        value={search}
        autoCapitalize="none"
      />
      {search.length > 0 && (
        <TouchableOpacity onPress={handlePressClearSearch}>
          <Image
            source={clearIcon}
            resizeMode="contain"
            style={styles.clearIcon}
          />
        </TouchableOpacity>
      )}
    </View>
  );
}
export default BarSearch;
