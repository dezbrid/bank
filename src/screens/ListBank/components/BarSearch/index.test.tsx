import 'react-native';
import React from 'react';

import BarSearch from './index';
import {render} from '@testing-library/react-native';

describe('<BarSearch />', () => {
  it('<BarSearch /> snapshot', () => {
    const tree = render(<BarSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
