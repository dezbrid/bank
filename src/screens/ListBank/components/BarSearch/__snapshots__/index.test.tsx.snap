// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarSearch /> <BarSearch /> snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 100,
        "flexDirection": "row",
        "height": 45,
        "margin": 20,
        "paddingLeft": 10,
        "paddingRight": 15,
      },
      Object {
        "borderColor": "#F1F1F1",
        "borderWidth": 1,
        "shadowColor": "#939393",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/screens/ListBank/components/BarSearch/assets/ic_search.png",
      }
    }
    style={
      Object {
        "height": 25,
        "marginRight": 10,
        "width": 25,
      }
    }
  />
  <TextInput
    autoComplete="off"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="Search"
    style={
      Object {
        "flex": 1,
      }
    }
    value=""
  />
</View>
`;
